src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,64 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    public static final int INDEX_SUPPLY = 0;
+    public static final int INDEX_BUY = 1;
+    public static final int INDEX_SUM_ARRAY_LENGTH = 2;
+    public static final int ARRAY_COUNT = 1;
+    public static final String SUPPLY = "supply";
+    public static final String BUY = "buy";
+    public static final String TOTAL = "result";
+    public static final String DELIMITER = "\\W+";
+    public static final String SEPARATOR = System.lineSeparator();
+
     public void getStatistic(String fromFileName, String toFileName) {
+        writeList(fileWork(readList(fromFileName)),toFileName);
+    }
+
+    private String readList(String fromFileName) {
```

Comment:
A lot of extra action:  you generate one string using stringBuilter, then you split it, to create another string.
return just List<String> here and work within createReport method
```suggestion
    private List<String> readList(String fromFileName) {
```

Reply:
I haven't studied "List" yet.